QualificationChecker/Controllers/UtilityClass.cs

```diff
@@ -0,0 +1,122 @@
+﻿using QualificationChecker.Models;
+
+namespace QualificationChecker.Controllers
+{
+    public abstract class UtilityClass
+    {
+        protected UtilityClass(Candidate candidate, Organization organization)
+        {
+            Candidate = candidate;
+            Organization = organization;
+        }
+
+        public Candidate Candidate { get; private set; }
+
+        public Organization Organization { get; private set; }
+
+        public Dictionary<int, List<OrgQuestion>> OrgAnswersDictionary
+        {
+            get
+            {
+                return Organization.OrgQuestions
+                                    .GroupBy(q => q.PositionId)
+                                    .ToDictionary(group => group.Key, group => group.ToList());
+            }
+        }
+
+    }
+    // fake incoming query call that an http request
+    public class MatchRequest
+    {
+        public Organization Organization { get; set; }
+        public List<Candidate> Candidates { get; set; }
+    }
+
+    public class CandidateUtilities : UtilityClass
+    {
+        public CandidateUtilities(Candidate candidate, Organization organization) : base(candidate, organization)
+        {
+        }
+
+        // Method to tell if a Canidate is intrested in the position
+        public bool IsInterested()
+        {
+            if (Candidate.InterestedPositionIds == null || !Candidate.InterestedPositionIds.Any())
+            {
+                return false;
+            }
+
+            return Candidate.InterestedPositionIds.Any(candidatePositionId =>
+                Organization.OrgQuestions.Any(OrgQuestion => OrgQuestion.PositionId == candidatePositionId));
+        }
+        // Method to make sure they meet the age requirement
+        public bool MeetsAgeRequirement()
+        {
+            return Candidate.Age >= Organization.MinAgeRequirement;
+        }
+
+        // Method to return if the answers match
+        private bool MatchAnswers(bool candidateAnswers, bool organizationAnswers)
+        {
+            // Compare each answer
+            return candidateAnswers == organizationAnswers;
+        }
+        // method too loop through to and append if answers match
+        public List<Candidate> CheckQuestions()
+        {
+            bool allAnswersMatch = true;
+            var QualifiedCandidates = new List<Candidate>();
+
+            foreach (var candidateAnswer in Candidate.CandidateQuestions)
```

Comment:
Variable is answer, but iterating through questions

Reply:
Also, this is the most common mistake we get ... iterating candidate answers, and not the organization answers.  I can be nefarious and submit a single answer, and if I pass, I am good, even if there are 100 organization answers.  Or, if you check the count, I can submit the same answer 100 times.  Long story short, use the correct source of truth.

Reply:
This is the most common error in submissions, iterating a dangerous source of truth, the candidate answers.  Org answer is THE source of truth, the thing we can trust.  We should start with that, so we can ensure EACH required answer exists.  Otherwise I can submit a single answer, and pass, even if there are 100 answers required.  Or I can submit 100 of the same question.